arch/arm/core/aarch32/cortex_m/debug.c

```diff
@@ -37,7 +37,8 @@ bool z_arm_debug_monitor_event_error_check(void)
 			"MATCHED flag should have been cleared on read.");
 
 		return true;
-	} else if (SCB->DFSR & SCB_DFSR_BKPT_Msk) {
+	}
+	if (SCB->DFSR & SCB_DFSR_BKPT_Msk) {
 		/* Treat BKPT events as an error as well (since they
```

Comment:
Will this move away from else if slow it down too much?

Reply:
No, because the compiler won't execute the second test because the first ends in an early return anyway.